net/http/authn/authn.go

```diff
@@ -72,8 +80,14 @@ func (a *API) Authenticate(req *http.Request) (*http.Request, error) {
 
 	// if there is no authentication at all, we return an "unauthenticated" error,
 	// which may be helpful when debugging
-	if len(X509Certs(ctx)) < 1 && err != nil {
-		return req, errors.New("unauthenticated")
+	if len(X509Certs(ctx)) < 1 && Token(ctx) == "" {
+		err := errors.New("unauthenticated")
+		if len(authnErrors) > 0 {
+			err = errors.WithDetailf(err, "Invalid credentials: %s", strings.Join(authnErrors, "; "))
```

Comment:
Oh, yeah, I was just going on the same principle as that comment, based on how it ultimately gets output:

```
Detail: Invalid credentials: invalid token: "jackattack"
```

vs

```
Detail: invalid credentials: invalid token: "jackattack"
```

It looks weird to have a capital letter in the middle of a sentence (after the first colon).